dashboard/src/components/SideMenu/NavLink.tsx

```diff
@@ -1,25 +1,40 @@
 import { FormattedMessage } from 'react-intl';
 
-import { AnchorHTMLAttributes } from 'react';
+import { Link, LinkProps } from '@tanstack/react-router';
 
 import { MessagesKey } from '@/locales/messages';
 
 import { NavigationMenuLink } from '../ui/navigation-menu';
 
-interface INavLink extends AnchorHTMLAttributes<HTMLAnchorElement> {
+/*interface INavLink extends AnchorHTMLAttributes<HTMLAnchorElement> {
   idIntl: MessagesKey;
   icon: JSX.Element;
-}
-
-const NavLink = ({ icon, idIntl, ...props }: INavLink): JSX.Element => (
-  <NavigationMenuLink asChild>
-    <a className="flex items-center no-underline hover:text-sky-500" {...props}>
-      <span className="mr-3">{icon}</span>
-      <span className="text-center text-sm">
-        <FormattedMessage id={idIntl} />{' '}
-      </span>
-    </a>
-  </NavigationMenuLink>
-);
+  to?: string;
+}*/
+
+type INavLink = LinkProps & {
+  idIntl: MessagesKey;
+  icon: JSX.Element;
+  href?: string;
+  as?: string;
+};
+
+const NavLink = ({ icon, idIntl, as, ...props }: INavLink): JSX.Element => {
+  const LinkElement = as || Link;
```

Comment:
nit: same thing, is more common to use `??` instead of `||`

Reply:
Also, why are you using this LinkElement? IMO it would be better to add a link and make it disabled or doing nothing when clicked, but I don't have the entire context

Reply:
If you click in Documentation option in sidemenu, you will see a redirect for \tree route. This isn't the expected behavior. This should redirect to an external page. By my tests, the Link component has this behavior. Because this, I use a dinamic link wrapper (Tanstack Link or HTML link).